src/config/mongoose_config_spec.erl

```diff
@@ -606,22 +630,34 @@ outgoing_pool_connection(<<"ldap">>) ->
                  <<"tls">> => #section{items = tls_items(),
                                        wrap = {kv, tls_options},
                                        process = fun ?MODULE:process_tls_sni/1}
-                }
+                },
+       format_items = map,
+       include = always,
+       defaults = #{<<"rootdn">> => "",
+                    <<"password">> => "",
+                    <<"encrypt">> => none,
+                    <<"servers">> => ["localhost"],
+                    <<"connect_interval">> => 10000}
       };
 outgoing_pool_connection(<<"rabbit">>) ->
     #section{
        items = #{<<"amqp_host">> => #option{type = string,
                                             validate = non_empty},
                  <<"amqp_port">> => #option{type = integer,
                                             validate = port},
-                 <<"amqp_username">> => #option{type = string,
+                 <<"amqp_username">> => #option{type = binary,
                                                 validate = non_empty},
-                 <<"amqp_password">> => #option{type = string,
+                 <<"amqp_password">> => #option{type = binary,
                                                 validate = non_empty},
                  <<"confirms_enabled">> => #option{type = boolean},
                  <<"max_worker_queue_len">> => #option{type = int_or_infinity,
                                                        validate = non_negative}
-                }
+                },
+       format_items = map,
+       include = always,
+       defaults = #{<<"amqp_port">> => 5672,
```

Comment:
I am not sure about this one... other such options are not specified explicitly, maybe omit this one as well? It seems to be an implicit default in the library.